package-res/resources/web/pentaho/data/filter/AbstractFilter.js

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../../lang/Base",
+  "../../util/arg",
+  "require"
+], function(Base, arg, require,  Or, And, Not) {
+  "use strict";
+
+  /**
+   * @name AbstractFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @abstract
+   * @amd pentaho/data/filter/AbstractFilter
+   *
+   * @classdesc This tree structure acts as a filter for a data set and is made up of:
+   *
+   * * Non-leaf nodes
+   * * Leaf nodes
+   *
+   * Each non-leaf node can be seen as a filter that acts upon its leaf nodes
+   * (the atomic predicates of the filtering) and aggregates the outcomes of each. In this
+   * way the structure is able to provide an intensional representation of a selection of
+   * the given data set.
+   *
+   * @description The `AbstractFilter` class is the abstract base class of
+   * classes that represent a data set filter.
+   *
```

Comment:
The title page, and the `@name` tag already identify the class.

What about:

@classdesc The (abstract) base class for filters that represent subsets of a particular {@link pentaho.data.Table} object.
A `filter` is an intensional representation of a subset of items in a collection.

The hierarchy spawned by this base class allows building filters by composition, in the form of a tree of filters.
When filtering a particular item in a data set, the leaf nodes evaluate the data and return a boolean that signals if the item belongs to the set defined by the filter.
Non-leaf nodes act as aggregators of the outcomes of other nodes (leaf or non-leaf).